functions-app/report-psychometrician/service/detections/detect-answer-responses-after-cutoff.js

```diff
@@ -0,0 +1,44 @@
+'use strict'
+
+const R = require('ramda')
+const RA = require('ramda-adjunct')
+const moment = require('moment')
+const jsFormat = 'Y-MM-DDTHH:mm:ss.SSSZ'
+
+const report = require('./report')
+const getQuestionTimerStartEvent = require('./get-question-timer-start-event')
+const getAnswer = require('./get-answer')
+const { getDuration } = require('../../../lib/date.service')
+
+const detectAnswerResponsesAfterCutoff = function (data) {
+  if (!RA.isPlainObj(data)) {
+    throw new TypeError('data should be an object')
+  }
+  const markedAnswers = R.pathOr([], ['markedAnswers', 'answer'], data)
+  if (!RA.isArray(markedAnswers)) { return }
+  if (R.length(markedAnswers) === 0) { return }
+  const answers = R.path(['checkPayload', 'answers'], data)
+  const audits = R.pathOr([], ['checkPayload', 'audit'], data)
+
+  const output = markedAnswers.map(ma => {
+    const questionTimerStartedEvent = getQuestionTimerStartEvent(audits, ma.questionNumber, ma.factor1, ma.factor2)
+    const questionTimerTimestamp = R.prop('momentTimestamp', questionTimerStartedEvent)
+    if (!moment.isMoment(questionTimerTimestamp)) { return }
+    const cutoff = moment(questionTimerTimestamp).add(6.1, 'seconds')
+    const answer = getAnswer(answers, ma.questionNumber)
+    if (answer && moment.isMoment(answer.momentTimestamp)) {
```

Comment:
why not `if (answer && moment.isMoment(answer.momentTimestamp) && answer.momentTimestamp.isAfter(cutoff))`
Rather avoid nesting?
You could alternatively return if answer is not moment type as guard clause.

Reply:
how would that be more readable?

Reply:
does that aid readability?  i would say no... :/

Reply:
Then how about splitting in two statements, one that covers the guard clause `!answer || !moment.isMoment(answer.momentTimestamp)` that returns and the other one that contains the currently nested statement?

Reply:
i just don't see an issue with how it currently is.  my perspective - if you understand what it does, and you can easily interpret the code, then the author has done a fair job.  

there is always more than one way to write these constructs, and if the change offers little value then i see no point in requesting it.  its not like what you are suggesting will optimise, nor aid maintainability.  Unless i'm overlooking something?

Reply:
Even though there's only 1 level of nesting in this block, I believe it's more readable and understandable if it's done with a guard clause.
I am going to approve as it has nothing to do with optimisation and it is rather my opinion since generally I am against nesting.
https://refactoring.guru/replace-nested-conditional-with-guard-clauses

Reply:
I've added the guard clause as I'm not opposed to it.

Reply:
cool, as long as everyone is happy :)